metricbeat/metricbeat.yml

```diff
@@ -21,6 +21,25 @@ metricbeat.config.modules:
 
 #==================== Elasticsearch template setting ==========================
 
+# Example config for templates
+# 
+# When `setup.template.templates` is configured, the old configuration is ignored.
+#setup.template.name: "metricbeat"
+#setup.template.pattern: "metricbeat*"
+#setup.template.fields: "${path.config}/fields.yml"
+#setup.template.enabled: false 
+#setup.template.overwrite: true
+setup.template.templates:
+  - name: "metricbeat-mix"
+    pattern: "*"
+    modules: "aerospike,apache,system"
+    overwrite: true
+  - name: "metricbeat-docker"
```

Comment:
If multiple templates are used, should we always postfix it by the version?

All the multi template options I think are great for the default cases in Beats and APM but if users want to use additional very advanced use cases, they should manually set it up. There are too many things that can go wrong.

Historically a lot of users have remove the version number from indices and templates which now becomes an issue during migrations so I would rather limit the options and flexibility we give the users by deafult.

Reply:
I would not introduce adding some auto-magical version to templates and change how the version in indices is added with this PR. It is already very closely related to ILM and general config changes. I suggest we discuss version handling completely separate.

Reply:
In the current ILM PR versions are added magically and the users can't remove them (on purpose).

Reply:
I am not a fan of auto magic either, but the version is really helpful when you do migration from beats version. 

@ruflin in the example config for the name we could use a string with field reference, so its not automagic but we give them the tips.

Reply:
+1 on giving them tips if it does not complicate things.

Not sure what you meant by "but the version is really help when you do migration from beats"? Sounds like we are on the same page users should not be able to remove the version?